x-pack/docs/en/security/authentication/security-domain.asciidoc

```diff
@@ -0,0 +1,99 @@
+users with the same `username`, it is reasonable to treat these users as the
+same end-user from a resource ownership point of view.
+
+With security domains, you group multiple realms to be under the same
+domain, end-users can authenticate with any of the realms and have access to the
+same set of resources regardless of which realm they authenticated with,
+(provided those realms are under the same domain).
+For example, a single <<user-profile,user profile>> is
+associated to the end-user, enabling preferences, notifications, and other
+user data to be shared across realms. The user can view results from an
+asynchronous search request or a scrolling search across realms. If the user has the
+necessary privileges, they can also view and manage API keys across realms.
+
+==== Configure a security domain
+
+You configure realms to be under a domain in `elasticsearch.yml`. The realms must
+be defined in `elasticsearch.yml` (but do not need to be enabled).
+
+NOTE: The <<file-realm,file realm>> and <<native-realm,native realm>> are
+automatically enabled as `default_file`, and `default_native`, respectively,
+without the need for explicit configuration. They can
+be listed under domains even when they are not explicitly defined in
+`elasticsearch.yml`.
+
+. Add a security domain configuration to `elasticsearch.yml` under the
+`xpack.security.authc.domains` namespace:
++
+[source, yaml]
+----
+xpack:
+  security:
+    authc:
+      domains:
+        my_domain:
+          realms: [ 'default_native', 'saml1' ] <1>
+----
+<1> This configuration defines a security domain called `my_domain`, which
+contains two realms named `default_native` and `saml1`.
+
+. Restart {es}.
+
+{es} can fail to start on invalid domain configuration including the followings:
+* A single realm is configured to be under multiple domains
+* Any undefined realm, or synthetic realm, or the reserved realm is configured to
+  be under a domain
+
+
+[NOTE]
+====
+Though it's possible to configure different <<roles,roles>> for the same user when
+authenticating with different realms, it is _not_ recommended.
+Using <<configuring-authorization-delegation,authorization delegation>>
+is a good way to ensure that a user is assigned the same roles from multiple realms.
+
+{es} always enforces the privileges associated with the currently authenticated user.
+Security domains don't bypass <<authorization,user authorization>> when
+resource sharing requires them. For example, a user requires the
+`manage_own_api_key` privilege to manage their own API keys. If that user does
+not have this privilege when authenticating with one realm, they won't be able
+to manage API keys while authenticating with this realm.
+This behaviour might be unexpected since the same user may or may not be able
+to perform the same operation depending on the realm it authenticates with.
+But it is a safe behaviour that ensures actual actions always comply to
+authorization requirements.
+====
```

Comment:
```suggestion
.Managing privileges across realms
***
Though it's possible to configure different <<roles,roles>> for the same user when
authenticating with different realms, it is _not_ recommended.
Using <<configuring-authorization-delegation,authorization delegation>>
is a good way to ensure that a user is assigned the same roles from multiple realms.

{es} always enforces the privileges associated with the currently authenticated user.
Security domains don't bypass <<authorization,user authorization>> when
resource sharing requires them. For example, a user requires the
`manage_own_api_key` privilege to manage their own API keys. If that user doesn't
have this privilege when authenticating with one realm, they won't be able
to manage API keys while authenticating with this realm.
This behavior might be unexpected since a single user may or may not be able
to perform the same operation depending on the realm it authenticates with.
However, it's a safe behavior that ensures actual actions always comply to
authorization requirements.
****
```